plugins/sync.js

```diff
@@ -457,37 +488,63 @@ window.plugin.sync.DataManager.prototype.createFile = function(callback) {
+//   onFileLoadedCallback        - function to handle success  
+//   handleError                 - function to handle Error
+window.plugin.sync.DataManager.prototype.readFile = function(needInitializeFileCallback,
+           onFileLoadedCallback,
+           handleError)   {
   var _this = this;
 
-  gapi.client.load('drive', 'v3').then(function () {
-    gapi.client.drive.files.get({ fileId: _this.fileId, alt: 'media' })
-    .then(function (response) {
-      var res = response.result;
-      if (res) {
-        onFileLoadedCallback(res);
-      } else {
-        needInitializeFileCallback();
-      }
-    },function(reason){
+  // gapi.client.load is a depricated routine
+  //   .then(opt_onFulfilled, opt_onRejected)
+  gapi.client.load('drive', 'v3').then(             // depricated, needs replacement
+    // gapi.client.drive.files.get
+    //   .then(opt_onFulfilled, opt_onRejected)
+    function ()                                     // opt_onFulfilled for .load
+      {gapi.client.drive.files.get({                // Gets a file's metadata by ID
+        fileId: _this.fileId,                       //   fileId
+        alt: 'media'                                //   respond with the file contents in 
+                                                    //   the response body
+      }) 
+      .then(
+        function (response) {                       // opt_onFulfilled for drive.files.get
+          var res = response.result;
+          if (res) {
+            onFileLoadedCallback(res);              // perform provided routine
+          } else {
+            needInitializeFileCallback();           // no result (data) returned
+          }
+        },
+        function(reason){                           // opt_onRejected for drive.files.get
+          handleError(reason);
+        }
+      );
+    },
+    function(reason){                               // opt_onRejected for .load
       handleError(reason);
-    });
-  },function(reason){
-    handleError(reason);
-  });
+    }
+  );
 };
 
+// saveFile
+//   data - the data to be saved
 window.plugin.sync.DataManager.prototype.saveFile = function(data) {
   var _this = this;
-
-  gapi.client.load('drive', 'v3').then(function () {
-    gapi.client.request({
-      path: '/upload/drive/v3/files/'+_this.fileId,
-      method: 'PATCH',
-      params: { uploadType: 'media' },
-      body: JSON.stringify(data)
-    }).execute();
-  });
+  
+  gapi.client.load('drive', 'v3').then(
+    function () {        
+      gapi.client.request({
+        path: '/upload/drive/v3/files/'+_this.fileId,
+        method: 'PATCH',                                    // why patch? 
+        params: { uploadType: 'media' },
+        body: JSON.stringify(data)
+      }).execute();                                         // write (patch) the file on drive
+    }
+                                                            // no errorhandling here?!
```

Comment:
again: don't ask me

if you think this needs rework/review add a "TODO" or "FIXME" in comment.
(and make sure to handle these TODOs really soon)